jcl/src/java.base/share/classes/com/ibm/tools/attach/target/IPC.java

```diff
@@ -261,7 +262,8 @@ static private native int createFileWithPermissionsImpl(String path,
 	static String getTmpDir() {
 		String tmpDir = getTempDirImpl();
 		if (null == tmpDir) {
-			tmpDir = com.ibm.oti.vm.VM.getVMLangAccess().internalGetProperties().getProperty("java.io.tmpdir"); //$NON-NLS-1$
+			IPC.logMessage("Could not get system temporary directory. Trying "+JAVA_IO_TMPDIR); //$NON-NLS-1$
+			tmpDir = com.ibm.oti.vm.VM.getVMLangAccess().internalGetProperties().getProperty(JAVA_IO_TMPDIR); //$NON-NLS-1$
```

Comment:
will this default to /tmp if not set?

Reply:
The system property is set by `getPlatformPropertyList()` to the value of the TMPDIR environment variable (sans character set conversion, which is the present problem).  If that does not resolve to a valid path, then it defaults to "/tmp".